### ISSUE TITLE
rules vue/script-indent behave differently with eslint rule indent

### ISSUE THREAD

**Issue by noe132** [vuejs/eslint-plugin-vue#443]
code
![tim 20180325164859](https://user-images.githubusercontent.com/2271900/37873301-79133398-304c-11e8-8bb8-8f66c51ace76.png)
it's valid indent applied by rule indent ([https://eslint.org/docs/rules/indent](https://eslint.org/docs/rules/indent))
but invalid with vue/script-indent ([https://github.com/vuejs/eslint-plugin-vue/blob/v4.4.0/docs/rules/script-indent.md](https://github.com/vuejs/eslint-plugin-vue/blob/v4.4.0/docs/rules/script-indent.md))

**Tell us about your environment**

* **ESLint Version:** 4.13.1
* **eslint-plugin-vue Version:** 4.4.0
* **Node Version:** 8.8.1

**Please show your full configuration:**

```
// http://eslint.org/docs/user-guide/configuring
const vueRules = {
  'vue/html-indent': ['error', 2, {
    'attribute': 1,
    'closeBracket': 0,
    'alignAttributesVertically': false,
    'ignores': []
  }],
  'vue/script-indent': ['error', 2, {
    'baseIndent': 1,
    'switchCase': 0,
    'ignores': []
  }],
  'vue/attribute-hyphenation': ['error', 'always'],
  'vue/mustache-interpolation-spacing': [2, 'always'],
  'vue/name-property-casing': ['error', 'PascalCase'],
  'vue/no-multi-spaces': 'error',
  'vue/require-default-prop': 'error',
  'vue/require-prop-types': 'error',
  'vue/v-bind-style': ['error', 'shorthand'],
  'vue/v-on-style': ['error', 'shorthand'],
  'vue/html-quotes': ['error', 'double'],
  'vue/no-confusing-v-for-v-if': 'error',
  'vue/this-in-template': ['error', 'never'],
  'vue/html-closing-bracket-newline': ['error', {
    'singleline': 'never',
    'multiline': 'never'
  }],
  'vue/html-closing-bracket-spacing': ['error', {
    'startTag': 'never',
    'endTag':  'never',
    'selfClosingTag': 'always',
  }],
}

const customRules = {
  // don't require .vue extension when importing
  'import/extensions': ['error', 'always', {
    'js': 'never',
    'vue': 'never'
  }],
  // allow optionalDependencies
  'import/no-extraneous-dependencies': ['error', {
    'optionalDependencies': ['test/unit/index.js']
  }],
  'consistent-return': 0,
  'function-paren-newline': ['error', 'consistent'],
  // allow debugger during development
  'object-curly-newline': ['error', { 'consistent': true }],
  'no-debugger': process.env.NODE_ENV === 'production' ? 2 : 0,
  'no-param-reassign': ['off'],
  'no-restricted-syntax': [
    'error',
    'ForInStatement',
    'LabeledStatement',
    'WithStatement',
  ],
  'prefer-destructuring': 'off',
}

module.exports = {
  root: true,
  parserOptions: {
    parser: 'babel-eslint',
    sourceType: 'module',
  },
  env: {
    browser: true,
  },
  extends: [
    'airbnb-base',
    'plugin:vue/essential',
  ],
  // required to lint *.vue files
  plugins: [
    'vue'
  ],
  // check if imports actually resolve
  'settings': {
    'import/resolver': {
      'webpack': {
        'config': 'build/webpack.base.conf.js'
      }
    }
  },
  // add your custom rules here
  rules: Object.assign({}, customRules, vueRules),
  globals: {
    fetch: true,
    window: true,
    FormData: true,
    Qiniu: true,
    Chart: true,
  },

  overrides: [
    {
      files: ['*.vue'],
      rules: {
        'indent': 'off',
        'max-len': 'off',
      }
    },
    {
      files: ['*.js'],
      rules: {
        'vue/script-indent': 'off',
      }
    }
  ]
}
```

**What did you do? Please include the actual source code causing the issue.**

```vue
<script>
  Promise.resolve()
    .then(() => {
      // SOME CODE HERE
    });
</script>
```

**What did you expect to happen?**
linting good with no error

**What actually happened? Please include the actual, raw output from ESLint.**
```
yarn run v1.3.2
$ C:\Users\noe\src\easypass-fe-hr\node_modules\.bin\eslint .\src\**\file.vue

C:\Users\**\file.vue
  301:1  error  Expected indentation of 10 spaces but found 12 spaces  vue/script-indent
  302:1  error  Expected indentation of 8 spaces but found 10 spaces   vue/script-indent

✖ 2 problems (2 errors, 0 warnings)
  2 errors, 0 warnings potentially fixable with the `--fix` option.
```

**Comment by Will-GitHub94** [comment]
I am also experiencing this issue *(similar one)*

Versions:

- eslint-plugin-vue: 4.4.0
- node: 8.9.1
- eslint: 4.18.2

Image below has error:

> Expected indentation of 9 tabs but found 8 tabs

![indent-fail](https://user-images.githubusercontent.com/7978646/37877035-da77f2e0-304c-11e8-905b-7b93ef7786d3.PNG)

This image has the error:

> Expected semicolon

![semi-fail](https://user-images.githubusercontent.com/7978646/37877047-0da52408-304d-11e8-93d9-3c93f95101cf.PNG)

**Comment by noe132** [comment]
by the way, this issue was not only got false positive error report with semicolon, also with trailing comma
```vue
<script>
  Promise.all([
    Promise.resolve()
      .then(() => {
      
      }), // Expected indentation of 4 spaces but found 6 spaces   vue/script-indent
  ])
</script>
```
and if remove that comma indent was fine but with missing trailing comma if trailing comma rule was set to always

**Comment by michalsnik** [comment]
Yeah, I confirmed it, seems that this line is causing that: https://github.com/vuejs/eslint-plugin-vue/commit/c6d25af2f1c02f0f1171edbacca51751f84153ee#r27776203
cc @mysticatea
